tests/robustness/robustness_test/main_test.go

```diff
@@ -48,20 +48,38 @@ func TestMain(m *testing.M) {
 	eng = th.engine
 
 	// Restore a random snapshot into the data directory
-	_, err := eng.ExecAction(ctx, engine.RestoreIntoDataDirectoryActionKey, nil)
-	if err != nil && !errors.Is(err, robustness.ErrNoOp) {
+	if _, err := eng.ExecAction(ctx, engine.RestoreIntoDataDirectoryActionKey, nil); err != nil && !errors.Is(err, robustness.ErrNoOp) {
 		th.cleanup(ctx)
 		log.Fatalln("Error restoring into the data directory:", err)
 	}
 
+	// Upgrade the repository format version if the env var is set
+	if os.Getenv("UPGRADE_REPOSITORY_FORMAT_VERSION") == "ON" {
+		log.Print("Upgrading the repository.")
+
+		err := th.upgrader.ConnectRepo("filesystem", "--path="+dataRepoPath)
+		exitOnError("failed to connect to repository", err)
+
+		rs, err := th.upgrader.GetRepositoryStatus()
+		exitOnError("failed to get repository status before upgrade", err)
+
+		prev := rs.ContentFormat.MutableParameters.Version
+
+		log.Println("Old repository format:", prev)
+		th.upgrader.UpgradeRepository(dataRepoPath)
+
+		rs, err = th.upgrader.GetRepositoryStatus()
+		exitOnError("failed to get repository status after upgrade", err)
+
+		curr := rs.ContentFormat.MutableParameters.Version
+		log.Println("Upgraded repository format:", curr)
+	}
```

Comment:
A bit late :) but could this be decomposed into a separate function to reduce complexity of this top-level function? Might be easier to read as a single line `maybeUpgradeRepoFormat()` or something

Reply:
Or you could even decompose the snapshot restore + upgrade into a `prepareRepository()` function, which calls `restoreRandomSnapshot()` and `maybeUpgradeRepoFormat()`, for even better simplification?